tests/test_audio_resampler.py

```
@@ -119,4 +119,78 @@ def test_single_sample_upsample():
+
+    # One-shot reference resample of the entire signal.
+    one_shot_out, _ = resample_audio(full_pcm, 8000, 24000)
+
+    # Chunked resample with state carried across irregular chunk sizes
+    # (not a clean multiple of the rate ratio) so the phase must be
+    # carried correctly to avoid drift/loss.
+    state = None
+    chunked_out = b""
+    offset = 0
+    while offset < len(full_pcm):
+        chunk_samples = random.choice([37, 53, 80, 113, 160])
```

Ruff (0.15.18)

[error] Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)